cas-server-core-authentication/src/main/java/org/jasig/cas/authentication/DefaultAuthenticationSupervisor.java

```diff
@@ -0,0 +1,110 @@
+package org.jasig.cas.authentication;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * This is {@link DefaultAuthenticationSupervisor}.
+ *
+ * @author Misagh Moayyed
+ * @since 4.2.0
+ */
+@Component("defaultAuthenticationSupervisor")
+public final class DefaultAuthenticationSupervisor implements AuthenticationSupervisor {
```

Comment:
That is possible; but there are a few concerns:

1. The authentication manager is already too complicated as it is. Way too many configuration knobs. 
2. The proposal seems like a mixture of responsibilities to me; an authentication manager handles the authentication piece of given credentials. The supervisor handles the collection of those credentials. I would prefer to keep things separate.